tests/rptest/tests/admin_uuid_operations_test.py

```diff
@@ -82,18 +95,251 @@ def _uuids_updated():
+                                 timeout_sec=30,
+                                 backoff_sec=1)
+
+        self.logger.debug(
+            f".. and decommission ghost node [{ghost_node_id}]...")
+        self.admin.decommission_broker(ghost_node_id, node=to_stop)
+
+        self.logger.debug(
+            "Check that all this state sticks across a rolling restart")
+
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            auto_assign_node_id=True)
+
+        self.redpanda.wait_until(self.redpanda.healthy,
+                                 timeout_sec=30,
+                                 backoff_sec=1)
+
+        def expect_ids(node: ClusterNode, uuid: str, id: int):
+            resp = self.admin.get_broker_uuid(node)
+            try:
+                assert resp[
+                    'node_id'] == id, f"Bad node id after override: '{resp['node_id']}', expected '{id}'"
+                assert resp[
+                    'node_uuid'] == uuid, f"Bad node uuid after override: '{resp['node_uuid']}', expected '{uuid}'"
+            except AssertionError as e:
+                self.logger.debug(e)
+                return False
+            return True
+
+        wait_until(lambda: expect_ids(to_stop, old_uuid, initial_to_stop_id),
+                   timeout_sec=30,
+                   backoff_sec=1,
+                   retry_on_exc=True)
+
+    @cluster(num_nodes=3)
+    @parametrize(mode=TestMode.CFG_OVERRIDE)
+    @parametrize(mode=TestMode.CLI_OVERRIDE)
+    def test_force_uuid_override_multinode(self, mode):
+        to_stop = self.redpanda.nodes[1:]
+        initial_to_stop_ids = [self.redpanda.node_id(n) for n in to_stop]
+
+        self.logger.debug("Kill one node, all is good")
+
+        self._restart_node(to_stop[0], drop_disk=True)
+
+        uuids = wait_until_result(
+            lambda: self._uuids_updated(),
+            timeout_sec=30,
+            backoff_sec=2,
+            err_msg="Node was unable to join the cluster")
+
+        ghost_node_id = self.admin.get_broker_uuid(to_stop[0])['node_id']
+
+        old_uuids = {}
+        for n in uuids:
+            id = n['node_id']
+            if id in initial_to_stop_ids:
+                old_uuids[id] = n['uuid']
+
+        assert len(
+            old_uuids) == 2, f"Unexpected old_uuids: {json.dumps(old_uuids)}"
+
+        self.logger.debug("Drop another node, this time restart should fail")
+
+        for n in to_stop:
+            with expect_exception(TimeoutError, lambda _: True):
+                self._restart_node(n, drop_disk=True)
+
+        current_uuids = [self.scrape_uuid(n) for n in to_stop]
+        assert len(current_uuids
+                   ) == 2, f"Missing some UUIDs: {json.dumps(current_uuids)}"
+
+        self.logger.debug(
+            "Restart both nodes again, with overrides. Keep both disks")
+
+        if mode == TestMode.CFG_OVERRIDE:
+            self.redpanda.restart_nodes(
+                to_stop,
```

Comment:
Nit: we are already kind of testing that the nodes will only adopt overrides that match the current uuid, but maybe it will be more realistic to do what k8s will do and perform a full rolling restart.

Reply:
Fair point. Will do

Reply:
I'm beginning to think that "rolling restart" (I used this language in the RFC and the runbook) is not quite accurate. In DT, for example, we have a `RollingRestarter` that a) uses maintenance mode by default b) requires the cluster to be healthy before cycling each node. Presumably a k8s rolling restart looks somewhat similar.

Of course we can't meet either of those in this usage case. In DT we can fudge it with an `unsafe` param or something, but we may want to reconsider this framing with respect to support-facing docs.

Reply:
It's worse than that, I think - in this test case, I believe a rolling restart is straightforwardly impossible, since restarting one node gives w/ overrides gives us a total of 2 live brokers - not enough for to form a cluster due to the presence of ghost nodes.

My intuition is that we can start the nodes concurrently (as written) as long as the number of nodes we're restarting is `<= (n_nodes + n_ghosts) / 2`. What matters is that the empty nodes don't form a cluster among themselves, independent of any nodes containing a complete controller log, right?

Reply:
Yeah, that makes sense to me. We want to restart the minimum number of ghost nodes to form a healthy cluster and wait for them to form a healthy controller quorum before continuing to restart the remaining nodes.

It makes sense to call out in the support docs not to use decommissioning or maintenance mode during any of this.

Perhaps the most important thing here is to not restart the healthy nodes to make sure they are available throughout the whole process.

Reply:
>most important thing here is to not restart the healthy nodes

yest exactly. in fact, if we restart those nodes, I think they will get stuck just the same.

>call out in the support docs not to use decommissioning or maintenance mode

Yup, and that they are unavailable.

Generally, I think we're back to the point where we have a stack of assumptions that look right but would benefit from a ✅  from @mmaslankaprv  when he becomes available.